containerd-shim-v2/service.go

```diff
@@ -350,7 +350,46 @@ func (s *service) ResizePty(ctx context.Context, r *taskAPI.ResizePtyRequest) (*
 
 // State returns runtime state information for a process
 func (s *service) State(ctx context.Context, r *taskAPI.StateRequest) (*taskAPI.StateResponse, error) {
```

Comment:
Just noticing the way this function is implemented brings duplication additionally to the states of the Kata API. Instead, it maintains its own states.
@bergwolf @lifupan WDYT?

Reply:
I thinks this states is needed. First the containerd requested states isn't entirely match to kata's states. there must be a transfer between them and this states does it; Second, there is some states value kata doesn't keep thus shimv2 needs to store it in this states.

Reply:
Yes I realized that, and I saw that you tried to translate the Kata states when possible, but I understand containerd expects more states than what we have with Kata. I'm not asking to change anything, it's just sad that we have to redefine another layer of states again...